lib/counter_culture/counter.rb

```diff
@@ -176,6 +201,12 @@ def first_level_relation_foreign_key
       relation_reflect(first_relation).foreign_key
     end
 
+    def polymorphic?
+      relation_reflect(relation).polymorphic?.tap do |is_polymorphic|
+        raise "Polymorphic associations only supported with one level" unless (relation.is_a?(Symbol) || relation.length == 1) if is_polymorphic
```

Comment:
I thought this was a necessary constraint -- but I never fully got my head around multi-levels. But let me think it through here:

You've got a 

```ruby
class Category; end
```

and
```ruby
class Employee; belongs_to :company; has_many :images, as: :imageable; end
```

and

```ruby
class Building; belongs_to :company; has_many :images, as: :imageable; end
```

and then a 

```ruby
class Image
  belongs_to :imageable, polymorphic: true
  # easy:
  counter_culture :imageable
  # possible (but why)
  counter_culture [:category, :imageable]
  # more likely, but still seems contrived
    counter_culture [:category, :imageable], :column_name => proc {|model| "#{model.imageable_type.downcase}_image_count" }

end
```

Ok -- that seems reasonable enough, as a test case. Now I guess I could try implementing it (might just work... already). If it gets tricky, I might just back out and leave this to someone who actually needs it.